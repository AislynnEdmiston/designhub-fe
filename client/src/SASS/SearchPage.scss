@import './mixins';
@import './palette';

.search-container {
  width: 100%;
  color: $white;
  margin-top: 15px;
  p {
    letter-spacing: 0.11em;
    font-size: 16px;
    font-style: normal;
    font-weight: normal;
    span {
      font-weight: bold;
    }
  }
  .search-tabs-container {
    background: $black;
    border-radius: 10px;
    margin-top: 20px;
    width: 100%;

    .tabs-container {
      height: 90vh;
      padding: 20px 30px;
    }
    .nav-links {
      display: flex;
      flex-flow: row nowrap;
      padding: 22px 36px;
      border-bottom: 1px solid $almost-black;
      .links {
        margin-right: 50px;
        letter-spacing: 0.11em;
        color: $dark-grey;
      }
      .active-link {
        color: $white;
      }
    }
  }
}

.search-user-card-container {
  display: flex;
  flex-flow: row nowrap;
  width: 300px;
  letter-spacing: 0.11em;
  height: 82px;
  align-items: center;
  margin-bottom: 40px;
  .user-card-avatar {
    img {
      width: 82px;
      height: 82px;
      border-radius: 100%;
      border: 3px solid $blue;
      margin-right: 20px;
      object-fit: cover;
    }
  }
  .user-card-info {
    h3 {
      font-size: 18px;
      line-height: 21px;
    }
    p {
      font-size: 12px;
      line-height: 14px;
    }
  }
}
