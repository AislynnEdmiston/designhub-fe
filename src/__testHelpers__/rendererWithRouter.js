import React from 'react'
import { Router } from 'react-router-dom';
import { Provider } from 'react-redux'
import { createStore, applyMiddleware } from 'redux'
import { createMemoryHistory } from 'history';
import renderer from 'react-test-renderer';
import thunk from 'redux-thunk';

import rootReducer from '../store/reducers';
jest.mock('../auth-wrapper');

const store = createStore(
  rootReducer,
  applyMiddleware(thunk)
);
const rendererWithRouter = (
  ui,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] })
  } = {}
) => {
  return {
    ...renderer.create(
      <Provider store={store}>
        <Router history={history}>
          {ui}
        </Router>
      </Provider>
    ),
    history
  }
}

export default rendererWithRouter;
