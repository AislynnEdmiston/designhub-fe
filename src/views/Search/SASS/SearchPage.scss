@import '../../../common/SASS/palette';
@import '../../../common/SASS/mixins';

.search-container {
  width: 100%;
  color: $white;
  margin-top: 15px;
  @include themify($themes) {
    color: themed('text');
  }
  p {
    letter-spacing: 0.11em;
    font-size: 16px;
    font-style: normal;
    font-weight: normal;
    span {
      font-weight: bold;
    }
  }
  .search-tabs-container {
    background: $black;
    border-radius: 10px;
    margin-top: 20px;
    width: 100%;
    @include themify($themes) {
      background: themed('bg2');
    }

    a {
      text-decoration: none;
      color: $white;
      @include themify($themes) {
        background: themed('bg2');
        color: themed('link');
      }
    }

    .tabs-container {
      min-height: 70vh;
      height: 100%;
      padding: 20px 30px;
      overflow-x: scroll;
      
      .results-container {
        height: 100%!important;
      }

      .empty {
        width: 100%;
        height: 100%;
        display: flex;
        justify-content: center;
        align-items: center;
        flex-direction: column;
        .empty-icon {
          user-select: none;
          opacity: 0.08;

          @include themify($themes) {
            filter: themed('filter');
          }
        }

        p {
          color: $dark-grey;
        }
      }

      .projects-array {
        display: grid;
        grid-template-columns: 22% 22% 22% 22%;
        grid-column-gap: 3%;
        width: 100%;

        .project-content {
          width: 100%;
          height: 14vw;
          // height: 11.5rem;
          position: relative;
          transition: all 0.2s ease-in-out;
          border: 8px solid $field-bg;
          border-radius: 5px;
          overflow: hidden;
          margin: 10px 10px 10px 10px;
          object-fit: cover;

          @include themify($themes) {
            border: 8px solid themed('bg');
          }

          .project-info {
            position: absolute;
            bottom: 0;
            width: 103%;
            color: $white;
            text-decoration: none;
            background: $blue;
            overflow: hidden;
            height: 50px;
            display: none;
            pointer-events: none;

            h1 {
              padding: 20px;
            }
          }

          .project-thumbnail {
            object-fit: cover;
            width: 103%;
            height: 100%;
            transition: 0.1s ease-in-out;
            cursor: pointer;
          }
        }
        .project-content:hover {
          border: 8px solid $blue;
        }
      }

      .project-content:hover {
        .project-info {
          display: inline;
          animation: slideUp 0.25s ease-in-out;
        }
      }

      @keyframes slideUp {
        from {
          height: 0px;
        }
        to {
          height: 50px;
        }
      }
    }
    .nav-links {
      display: flex;
      flex-flow: row nowrap;
      padding: 22px 36px;
      border-bottom: 1px solid $almost-black;
      .links {
        margin-right: 50px;
        letter-spacing: 0.11em;
        color: $dark-grey;
        cursor: pointer;
      }
      .active-link {
        color: $white;
        @include themify($themes) {
          color: themed('text');
        }
      }
    }
  }
}

.search-user-card-container {
  display: flex;
  flex-flow: row nowrap;
  box-sizing: border-box;
  width: 300px;
  letter-spacing: 0.11em;
  height: 82px;
  align-items: center;
  margin-bottom: 40px;
  .user-card-avatar {
    img {
      width: 82px;
      height: 82px;
      border-radius: 100%;
      border: 3px solid $blue;
      margin-right: 20px;
      object-fit: cover;
    }
  }
  .user-card-info {
    h3 {
      font-size: 18px;
      line-height: 21px;
    }
    p {
      font-size: 12px;
      line-height: 14px;
    }
  }
}
