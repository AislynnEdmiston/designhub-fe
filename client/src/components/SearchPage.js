import React, { Component } from 'react';
import { Tab, Tabs, TabList, TabPanel } from 'react-tabs';
import { Link } from 'react-router-dom';
import defaultImg from '../ASSETS/default_thumbnail.svg';

import empty from './Icons/empty_project.svg';

import SearchUserCard from './SearchUserCard';

import '../SASS/SearchPage.scss';
import '../SASS/UserProfile.scss';

class SearchPage extends Component {
  displayUsers = () => {
    return this.props.searchData.users.map((user, index) => {
      return <SearchUserCard user={user} key={index} />;
    });
  };

  displayProjects = () => {
    return this.props.searchData.projects.map((project, index) => {
      return (
        <div className="project-content" key={index}>
          <Link to={`/project/${project.id}`}>
            <>
              <div className="project-info">
                <h1>{project.name}</h1>
              </div>
              <img
                src={project.mainImg ? project.mainImg : defaultImg}
                className="project-thumbnail"
                alt="test"
                key={project.id}
              />
            </>
          </Link>
        </div>
      );
    });
  };

  render() {
    const noProjects =
      !this.props.searchData.projects || !this.props.searchData.projects.length;
    const noUsers =
      !this.props.searchData.users || !this.props.searchData.users.length;

    return (
      <div className="search-container">
        <p>
          results for <span>{this.props.searchTerm}</span>
        </p>
        <div className="search-tabs-container">
          <Tabs>
            <TabList className="nav-links">
              <Tab className="links" selectedClassName="active-link">
                Projects
              </Tab>
              <Tab className="links" selectedClassName="active-link">
                Users
              </Tab>
            </TabList>
            <div className="tabs-container">
              <TabPanel className='results-container'>
                {noProjects && (
                  <div className="empty">
                    <img alt='empty' className="empty-icon" src={empty} />
                    <p>No projects found!</p>
                  </div>
                )}
                <div className="projects-array">
                  {this.props.searchData.projects && this.displayProjects()}
                </div>
              </TabPanel>
              <TabPanel className='results-container'>
                {noUsers && (
                  <div className="empty">
                    <img alt='empty' className="empty-icon" src={empty} />
                    <p>No users found!</p>
                  </div>
                )}
                <div className="users-array">
                {this.props.searchData.users && this.displayUsers()}
                </div>
              </TabPanel>
            </div>
          </Tabs>
        </div>
      </div>
    );
  }
}

export default SearchPage;
