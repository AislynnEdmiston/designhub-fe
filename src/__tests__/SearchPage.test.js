import React from 'react';
import SearchPage from '../views/Search/SearchPage';
import rendererWithRouter from '../__testHelpers__/rendererWithRouter';


describe('<SearchPage />', () => {

  describe('should match the previous snapshot of the component', () => {
    it('tests that the component has not changed and matched the previous snapshot', () => {
      const searchData = {
        id: 1
      }
      const tree = rendererWithRouter(
        <SearchPage searchData={searchData} history={{ location: { pathname: '/' } }} />
      );
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
})
