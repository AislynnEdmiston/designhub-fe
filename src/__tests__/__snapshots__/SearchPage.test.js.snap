// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchPage /> should match the previous snapshot of the component tests that the component has not changed and matched the previous snapshot 1`] = `
<div
  className="search-container"
>
  <p>
    results for 
    <span />
  </p>
  <div
    className="search-tabs-container"
  >
    <div
      className="react-tabs"
      data-tabs={true}
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      <ul
        className="nav-links"
        role="tablist"
      >
        <li
          aria-controls="react-tabs-1"
          aria-disabled="false"
          aria-selected="true"
          className="links active-link"
          id="react-tabs-0"
          role="tab"
          tabIndex="0"
        >
          Projects
        </li>
        <li
          aria-controls="react-tabs-3"
          aria-disabled="false"
          aria-selected="false"
          className="links"
          id="react-tabs-2"
          role="tab"
          tabIndex={null}
        >
          Users
        </li>
      </ul>
      <div
        className="tabs-container"
      >
        <div
          aria-labelledby="react-tabs-0"
          className="results-container react-tabs__tab-panel--selected"
          id="react-tabs-1"
          role="tabpanel"
        >
          <div
            className="empty"
          >
            <img
              alt="empty"
              className="empty-icon"
              src="empty_project.svg"
            />
            <p>
              No projects found!
            </p>
          </div>
          <div
            className="projects-array"
          />
        </div>
        <div
          aria-labelledby="react-tabs-2"
          className="results-container"
          id="react-tabs-3"
          role="tabpanel"
        />
      </div>
    </div>
  </div>
</div>
`;
